qiskit_dynamics/models/lindblad_models.py

```diff
@@ -127,6 +130,71 @@ def from_hamiltonian(
         return cls(
             hamiltonian_operators=hamiltonian.operators,
             hamiltonian_signals=hamiltonian.signals,
-            noise_operators=noise_operators,
-            noise_signals=noise_signals,
+            dissipator_operators=dissipator_operators,
+            dissipator_signals=dissipator_signals,
+            drift=hamiltonian.drift,
         )
+
+    @property
+    def frame(self):
+        return self._frame
+
+    @frame.setter
+    def frame(self, frame: Union[Operator, Array, Frame]):
+        if self._frame is not None and self._frame.frame_diag is not None:
+            self._operator_collection.drift = self._operator_collection.drift + Array(
+                np.diag(self._frame.frame_diag)
+            )
+            self._operator_collection.apply_function_to_operators(
+                self.frame.operator_out_of_frame_basis
+            )
+
+        self._frame = Frame(frame)
+
+        if self._frame.frame_diag is not None:
+            self._operator_collection.apply_function_to_operators(
+                self.frame.operator_into_frame_basis
+            )
+            self._operator_collection.drift = self._operator_collection.drift - Array(
+                np.diag(self._frame.frame_diag)
+            )
+
+    def evaluate_without_state(self, time: float, in_frame_basis: Optional[bool] = False) -> Array:
+        raise NotImplementedError("Lindblad models cannot be represented without a given state without vectorization.")
+
+    def evaluate_with_state(
+        self, time: Union[float, int], y: Array, in_frame_basis: Optional[bool] = False
+    ) -> Array:
+        """Evaluates the Lindblad model at a given time.
+        time: time at which the model should be evaluated
+        y: Density matrix as an (n,n) Array
+        in_frame_basis: whether the density matrix is in the
+            frame already, and if the final result
+            is returned in the frame or not."""
+
+        if not in_frame_basis:
+            y = self.frame.operator_into_frame_basis(y)
+
+        hamiltonian_sig_vals = np.real(self._hamiltonian_signals.complex_value(time))
+        dissipator_sig_vals = np.real(self._dissipator_signals.complex_value(time))
+
+        # Need to check that I have the differences chosen correctly
+        if self.frame.frame_diag is not None:
```

Comment:
This should be implemented purely by making calls to methods of `self._frame` (see other comment related to this).